content/src/service/synchronization/ContentCluster.ts

```diff
@@ -175,22 +186,50 @@ export class ContentCluster {
     }
 
     /** Detect my own identity */
-    private async detectMyIdentity(servers: Set<ServerMetadata>): Promise<void> {
+    async detectMyIdentity(attempts: number = 1): Promise<void> {
         try {
-            // Ask each server for their name
-            const serverNames = await Promise.all(Array.from(servers)
-                .map(async serverMetadata => ({ metadata: serverMetadata, name: await this.getServerName(serverMetadata.address) })))
-
-            // Filter out other servers
-            const serversWithMyName = serverNames.filter(({ name }) => name == this.nameKeeper.getServerName())
-
-            if (serversWithMyName.length > 1) {
-                ContentCluster.LOGGER.warn(`Expected to find only one server with my name '${this.nameKeeper.getServerName()}', but found ${serversWithMyName.length}`)
-            } else {
-                this.myIdentity = serversWithMyName[0]?.metadata
+            // Fetch server list from the DAO
+            const allServers: Set<ServerMetadata> = await this.dao.getAllContentServers()
```

Comment:
We should try to avoid execute this method as much as possible. Each call consume resources that we are paying.
Maybe we should add a cache for this and ensure to refresh it every 5 minutes, so we can call this method without problems.